### PR TITLE
Replaces ARMAs helicopter drag with a custom solution

### PR DESCRIPTION
**Pull request by BradMick**
Tuned to provide Max Range airspeed at the required torque per the PERF page.

### CODE REVIEW

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/config/cfgVehicles/sfmplus.hpp [hunk 1]
```diff
@@ -2,18 +2,18 @@ class Fza_SfmPlus {
...
     stabPos[]       = {0.0, -7.207, -0.50};  //m
     stabWidth       = 3.22;                  //m
     stabLength      = 1.07;                  //m
     centerOfMass[]  = {0.0, 2.08, 0.00};     //m
 
     //Fuel data
-    initFuelFracRobbie   = 0.39;
-    initFuelFracNoRobbie = 0.22;
+    initFuelFracRobbie   = 1.00;
+    initFuelFracNoRobbie = 1.00;
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/config/cfgVehicles/sfmplus.hpp:16]: Please can you undo these?

File: @AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/functions/core/fn_coreUpdate.sqf [hunk 2]
```diff
@@ -19,11 +19,26 @@ Author:
...
+private _exp               = -GRAVITY * MOLAR_MASS_OF_AIR * (_altitude - _referenceAltitude) / (UNIVERSAL_GAS_CONSTANT * (_temperature + DEG_C_TO_KELVIN));
+private _pressure          = ((_referencePressure / 0.01) * (EXP _exp)) * 0.01;
+
+private _densityAltitude   = (_altitude + ((SEA_LEVEL_PRESSURE - _altimeter) * 1000)) + (120 * (_temperature - (STANDARD_TEMP - ((_altitude / 1000) * 2))));
+private _dryAirDensity     = (_pressure / 0.01) / (287.05 * (_temperature + DEG_C_TO_KELVIN));
+
 //Set the center of mass
-private _forcePos = _heli getVariable "fza_sfmplus_centerOfMass";
-_heli setCenterofMass [_forcePos];
+//private _forcePos = _heli getVariable "fza_sfmplus_centerOfMass";
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/functions/core/fn_coreUpdate.sqf:40]: Rather than commenting this out, please could we delete these 2 lines and remove the comment?

File: @AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/functions/fuselage/fn_fuselageDrag.sqf [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+
+private _vecX = [1.0, 0.0, 0.0];
+private _vecY = [0.0, 1.0, 0.0];
+private _vecZ = [0.0, 0.0, 1.0];
+
+[velocityModelSpace _heli # 0, velocityModelSpace _heli # 1, velocityModelSpace _heli # 2] 
+    params ["_locVelX", "_locVelY", "_locVelZ"];
+
+private _dragX = -_fuselageDragCoefX * 0.5 * _rho * _fuselageAreaSide   * (_locVelX * _locVelX);
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/functions/fuselage/fn_fuselageDrag.sqf:39]: While it is an equivalent approach, could it be simpler to approach this using a singular drag force vector rather than 3 - one for each dimension? We should be able to get some 'SIMD' performance improvements using vector functions in SQF more.
```sqf
private _drag =
    [ _fuselageDragCoefX * _fuselageAreaSide * (_locVelX * _locVelX)
    , _fuselageDragCoefY * _fuselageAreaFront * (_locVelY * _locVelY)
    , _fuselageDragCoefZ * _fuselageAreaBottom * (_locVelX * _locVelZ)
    ] vectorMultiply (-0.5 * _rho * _deltaTime);

_heli addForce [_heli vectorModelToWorld _drag, _aerodynamicCenter];
```

File: @AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/headers/core.hpp [hunk 4]
```diff
@@ -10,4 +10,15 @@
 #define CENTRAL_ASIA_WINTER 5
 #define ASIA 6
 
-#define MIN_TIME_BETWEEN_UPDATES 0.3
\ No newline at end of file
+#define MIN_TIME_BETWEEN_UPDATES 0.3
+
+#define EPSILON                 0.000001
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/headers/core.hpp:15]: Nice use of constants

**Review by mattysmith22 — CHANGES_REQUESTED** [review]
Looks good so far! No correctness issues, just would like to not change the default fuel levels of the heli and make some simplicity improvements to the drag func

File: @AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/functions/fuselage/fn_fuselageDrag.sqf [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+private _forceZ = _vecZ vectorMultiply (_dragZ * _deltaTime);
+
+_heli addForce[_heli vectorModelToWorld _forceX, _aerodynamicCenter];
+_heli addForce[_heli vectorModelToWorld _forceY, _aerodynamicCenter];
+_heli addForce[_heli vectorModelToWorld _forceZ, _aerodynamicCenter];
+
+#ifdef __A3_DEBUG__
+private _colorRed = [1,0,0,1]; private _colorGreen = [0,1,0,1]; private _colorBlue = [0,0,1,1]; private _colorWhite = [1,1,1,1];
+
+DRAW_LINE = {
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/functions/fuselage/fn_fuselageDrag.sqf:57]: Please can you underscore (_) the beginning of this to make it private?

File: @AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/functions/aero/fn_aeroStabilator.sqf [hunk 6]
```diff
@@ -34,7 +34,7 @@ DRAW_LINE = {
 private _objCtr  = _heli selectionPosition ["modelCenter", "Memory"];
 private _stabPos = _heli getVariable "fza_sfmplus_stabPos";
 private _stabPvt = _objCtr vectorAdd _stabPos;
-
+
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/functions/aero/fn_aeroStabilator.sqf:37]: Please could you remove this tab here?

File: @AH-64D Apache Longbow/Addons/fza_ah64_systems/headers/systems.hpp [hunk 7]
```diff
@@ -21,7 +21,7 @@
 #define SYS_MIN_ACC_PSI   1650
 #define SYS_HYD_MIN_LVL   0.1
  
-#define SYS_BATT_TIMER    3.0   //min
+#define SYS_BATT_TIMER    12.0  //min
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_systems/headers/systems.hpp:24]: This should really be a separate PR due to it not being on-topic, but will ignore it this time

**Review by mattysmith22 — CHANGES_REQUESTED** [review]

File: @AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/functions/aero/fn_aeroStabilator.sqf (hunk 6 above)
Comment (BradMick) [reply @AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/functions/aero/fn_aeroStabilator.sqf:37]: Fixed

**Review by mattysmith22 — APPROVED** [review]
Thanks for the changes!
